src/app/src/main/java/com/example/bookmark/ScanIsbnActivity.java

```diff
@@ -1,18 +1,226 @@
+import com.google.mlkit.vision.barcode.BarcodeScanner;
+import com.google.mlkit.vision.barcode.BarcodeScannerOptions;
+import com.google.mlkit.vision.barcode.BarcodeScanning;
+import com.google.mlkit.vision.common.InputImage;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
 
 /**
- * TODO: Description of class.
+ * This activity opens up a camera view using CameraX and allows a
+ * user to scan a barcode using Google's ML kit. Upon success
+ * the isbn will be returned.
+ *
+ * TODO: Add more to these? Classes/Listeners? More about libraries used?
  * @author Mitch Adam.
  */
 public class ScanIsbnActivity extends AppCompatActivity {
+    // TODO: Figure out the back navigation
+
+    private Executor executor = Executors.newSingleThreadExecutor();
+    private int REQUEST_CODE_PERMISSIONS = 1001;
+    private final String[] REQUIRED_PERMISSIONS = new String[]{"android.permission.CAMERA"};
+
+    PreviewView mPreviewView;
+    BarcodeScanner scanner;
+
+    // Used to compare barcode readings to increase accuracy
+    int numRequiredSameBarcodes = 5;
+    int barcodesIndex = 0;
+    String[] barcodes = new String[numRequiredSameBarcodes - 1];
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_scan_isbn);
+        getSupportActionBar().setTitle("Scan ISBN");
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        mPreviewView = findViewById(R.id.viewFinder);
+
+        BarcodeScannerOptions options =
+            new BarcodeScannerOptions.Builder()
+                .setBarcodeFormats(
+                    Barcode.FORMAT_EAN_13, // Format of ISBN barcode's
+                    Barcode.FORMAT_EAN_8)
+                .build();
+        scanner = BarcodeScanning.getClient();
+
+        if(allPermissionsGranted()){
+            startCamera(); //start camera if permission has been granted by user
+        } else {
+            Log.d("PHOTO", "Should request permissions");
+            ActivityCompat.requestPermissions(this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS);
+        }
+    }
+
+    private void startCamera() {
+
+        final ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(this);
+
+        cameraProviderFuture.addListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+
+                    ProcessCameraProvider cameraProvider = cameraProviderFuture.get();
+                    bindPreview(cameraProvider);
+
+                } catch (ExecutionException | InterruptedException e) {
+                    // No errors need to be handled for this Future.
+                    // This should never be reached.
```

Comment:
If this should never be reached, we should get rid of the try/catch block. That way we will notice if an error does somehow occur.